### PR TITLE
MN-589: utility app

### PR DESCRIPTION
**Pull request by Hirama**
**- What I did**
A CLI app for creating an additional deposit in the member's wallet
**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

**Review by pavlushev — APPROVED** [review]

File: application/cmd/integrum/README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    Examples:
+    ./bin/integrum http://localhost:19001/admin-api/rpc -p http://localhost:19101/api/rpc -k 'Path to a key pair' -u postgresql://localhost:5432/postgres?sslmode=disable -v"
+    
+    Flags:
+        -u, --dbUrl string        URL with credential to the DB
+        -h, --help                Help for integrum
+        -k, --memberkeys string   Path to a key pair
+        -p, --publicUrl string    URL to the public Insolar API
+        -v, --verbose             Print request information
+        -w, --waitTime int        Milliseconds to wait after each transfer (default 500)
```
Comment (megge-dream) [inline application/cmd/integrum/README.md:48]:
```suggestion
        -w, --waitTime int        Milliseconds to wait after each deposit creation (default 500)
```

File: application/cmd/integrum/README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    Insolar integrum is a simple CLI tool for creating an additional deposit in the member's wallet at Insolar Platform
+    
+    Usage:
+      integrum <insolar_endpoint> [flags]
+    
+    Examples:
+    ./bin/integrum http://localhost:19001/admin-api/rpc -p http://localhost:19101/api/rpc -k 'Path to a key pair' -u postgresql://localhost:5432/postgres?sslmode=disable -v"
+    
+    Flags:
+        -u, --dbUrl string        URL with credential to the DB
```
Comment (megge-dream) [inline application/cmd/integrum/README.md:43]: Is it observer db? Its not clear from description

**Review by megge-dream — CHANGES_REQUESTED** [review]

**Review by megge-dream — APPROVED** [review]

File: application/cmd/integrum/main.go [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+				"WHERE d.eth_hash != 'genesis_deposit' AND d.eth_hash != 'genesis_deposit2' "+
+				"AND d.eth_hash NOT IN (SELECT d1.eth_hash FROM deposits d1 WHERE d1.eth_hash like '%_2');")
+			if err != nil {
+				panic(err)
+			}
+
+			if rows.RowsAffected() == 0 {
+				return errors.New("Members without second deposit not found")
+			}
+			fmt.Println(rows.RowsAffected())
```
Comment (uhzxl) [inline application/cmd/integrum/main.go:221]: Looks like the query doesn't work as expected.

When running a query on test data, I got this
image


```suggestion
			err = db.Model().
				Table("deposits").
				Column("deposit_ref", "member_ref").
				Where("status = 'confirmed'").
				Where("eth_hash != 'genesis_deposit'").
				Where("eth_hash != 'genesis_deposit2'").
				Where("eth_hash not like E'%\\_2'").
				Where(`not exists (
					select 1 
					from deposits as inn
					where inn.eth_hash = concat("deposits".eth_hash, '_2')
				)`).
				Select(&members)
			if err != nil {
				panic(err)
			}

			if len(members) == 0 {
				return errors.New("Members without second deposit not found")
			}
			fmt.Println(len(members))
```

**Review by uhzxl — CHANGES_REQUESTED** [review]
Would be better to extract main logic to somewhere (maybe to another file) cause now it mixed with input parameters processing and makes it difficult to understand

**Review by uhzxl — APPROVED** [review]

**Pull request merged**